### ISSUE TITLE
Gaining Insights from Fine-Tuned Model

### ISSUE THREAD

**Issue by vincentabraham** [meta-llama/llama#1061]
I have fine-tuned the Llama2-13b-chat-hf model for a binary classification problem. I'm getting a pretty good accuracy with testing the assistant prompt on a test dataset but is there a way with which I could ask the model, after it gives out the binary output, to tell me how did it come to that conclusion? So far, I've tried appending the assistant response to the string and appending another user prompt which asks for insights on the prediction. But this just gives me a garbage output (either a binary output again or just repeating the user prompt). Is there any way that I could do this and is fine-tuning even the right way to do this?

## Runtime Environment
- Model: `llama-2-13b-chat`
- Using via huggingface?: yes
- OS: Ubuntu
- GPU VRAM: 
- Number of GPUs:
- GPU Make: Nvidia

**Comment by subramen** [comment]
You might be able to use Chain-of-Thought prompting to get the model to generate its rationale as well. See the Chain of Thought section in this notebook: https://github.com/subramen/llama-recipes/blob/add-readmes/recipes/quickstart/Prompt_Engineering_with_Llama_2.ipynb 

An article I found that might help: https://medium.com/@JerryCuomo/lets-think-step-by-step-advanced-reasoning-in-business-with-chain-of-thought-prompting-dd5ae8a6008

**Comment by vincentabraham** [comment]
Thanks a lot for the suggestions! I'll try it out and let you know.  

> Instead of finetuning, have you considered providing a few examples of the task in the prompt (see "Few-Shot Prompting" in the notebook I linked above)? If your finetuned model has overfit it might not be able to generate a chain of thought.
The issue here is that the problem I'm using the model to solve involves a lot of numerical data and a binary classification based on the values in the data. I want the model to observe a pattern in the overall training dataset and use it to make predictions. And this is why I'm using fine-tuning (please let me know if there's a better way to solve my problem). The fine-tuned model achieves a good testing accuracy but it hallucinates (generates garbage numerical data with its answer) when I try to ask the model insights on how it made those predictions. Based on your response, it seems that the model is overfitted. So, is there any way to ensure that it does not happen?

**Comment by vincentabraham** [comment]
Thank you for the help.

**Comment by JhonDan1999** [comment]
hi @subramen 
> Instead of finetuning, have you considered providing a few examples of the task in the prompt (see "Few-Shot Prompting" in the notebook I linked above)? If your finetuned model has overfit it _might_ not be able to generate chain of thought.



This is very interesting I just experiment with a task of multiple choice question answering like:
<img width="715" alt="Screenshot 2024-05-29 at 3 52 03 AM" src="https://github.com/meta-llama/llama/assets/145554661/69b0dbbe-3d5a-494d-8e55-60ba8002e9b5">


I noticed that the model was not able to generate the expected answer unless I give it a few shot (one example) and I'm confused why the model was not able to stop such a simple task without any example do you have any insight in this ?
